src/database/database.go

```diff
@@ -44,23 +46,39 @@ func DSN(ci PostgreSQLInfo) string {
 		"password=%s dbname=%s sslmode=disable",
 		ci.Hostname, ci.Port, ci.Username, ci.Password, ci.DatabaseName)
 }
+
+//Data Source Name for Redis
 func DSN_Redis(ci RedisInfo) string {
 	return fmt.Sprintf("%s:%d", ci.URL, ci.Port)
 }
 
+//Setup Postgres Connection
 func SetupPostgres(d Info) (*sql.DB, error) {
-	db, err := sql.Open("postgres", DSN(d.PostgreSQL))
-	//check if is alive
-	err = db.Ping()
-	return db, err
+	if IsPostgresConnected == false {
+		db, err := sql.Open("postgres", DSN(d.PostgreSQL))
+		//check if is alive
+		err = db.Ping()
+		SetPostgresConnected()
+		log.Println("new connection to postgres")
+		return db, err
+	}
+	var err error
+	log.Println("reusing connection postgres connection ")
+	return DB, err
 }
 
 func SetupRedis(d Info) (redis.Conn, error) {
-	var err, pool = newPool(d)
-	connection := pool.Get()
-	return connection, err
+	if IsRedisConnected == false {
+		var err, pool = newPool(d)
+		connection := pool.Get()
+		SetRedisConnected()
```

Comment:
If you have method SetRedisConnected, than you should have CheckRedisConnected...

Or just write IsRedisConnected = true here

Pls, be consistent